hat/assets/js/apps/Iaso/domains/instances/hooks/speeddials.ts

```diff
@@ -23,20 +24,24 @@ export const useLinkOrgUnitToReferenceSubmission = ({
 }: LinkToFormParams): ((
     // eslint-disable-next-line no-unused-vars
     instance: Instance,
-    // eslint-disable-next-line no-unused-vars
-    isOrgUnitAlreadyLinked: boolean,
 ) => any) => {
     const { mutateAsync: saveOrgUnit } = useSaveOrgUnit(undefined, [
         'orgUnits',
     ]);
     const dispatch = useDispatch();
     return useCallback(
-        (currentInstance: Instance, isOrgUnitAlreadyLinked: boolean) => {
-            const { org_unit: orgUnit, id: instanceId } = currentInstance ?? {};
-            const id = isOrgUnitAlreadyLinked ? null : instanceId;
+        (currentInstance: Instance) => {
+            const {
+                org_unit: orgUnit,
+                id: instanceId,
+                is_reference_instance: isReferenceInstance,
+            } = currentInstance ?? {};
```

Comment:
Not sure this will work if `currentInstance` is empty has you are trying to get org_unit, id, ... of an empty object